app/models/user.rb

```diff
@@ -230,11 +230,8 @@ def email_domain
     nil
   end
 
-  def update_sign_in!(new_sign_in: false)
-    old_current = current_sign_in_at
-    new_current = Time.now.utc
-
-    self.last_sign_in_at     = old_current || new_current
+  def update_sign_in!(new_sign_in: false, new_current: Time.now.utc)
```

Comment:
Sorry, I do not understand this change. Why introduce a new parameter, when it is not being used anywhere? Just to save on LOC? That seems weird.

Now that it *is* there, you risk it being used in the future. So the question becomes, is there a valid use case for this? Maybe there is, but I cannot think of one right now. I *can* think of ways to use this to break assumptions, corrupt data and lead to very surprising behavior, though.

Overall, I do not agree that this change is worthwile. Sorry.